core/tests/protractor_utils/LibraryPage.js

```diff
@@ -77,6 +79,11 @@ var LibraryPage = function() {
       await searchInputs.first());
     await searchInput.clear();
     await searchInput.sendKeys(searchQuery);
+    let currentUrl = await browser.getCurrentUrl();
+    if (currentUrl.includes('/learn')) {
```

Comment:
This seems like too loose of a condition. For example, this would match a page with a user's profile page if their username were `learner123`. What if we decided whether to click the search button based on a parameter to the function instead?

This drew my attention to the fact that in the classroomPage test, we call libraryPage code while still on the classroom page:

https://github.com/oppia/oppia/blob/b314627a54b9ef7adcae06ab0e48efef3735982d/core/tests/protractor_desktop/classroomPage.js#L56-L57

I think we can fix both these issues at once. We first add a parameter to _submitSearchQuery and findExploration to specify whether to click the search button. It can default to false. Then we create a function in ClassroomPage.js that wraps findExploration but passes `true` to the search button parameter.

Alternatively:
* We could look for an HTML class on the classroom page to decide whether to click the search button
* We could do a stricter matching on the current URL. For example, ensure the `/learn` comes immediately after the domain. We'll have to do some more sophisticated parsing though to handle the localhost:8181 and 127.0.0.1:8181 cases on developer machines

Reply:
Yup, makes sense. I've added an explicit check for the button. PTAL!

Reply:
Thanks! all that's left is the flakiness reruns then